main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MissingLevelIdValidator.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.util.HashSet;
+import java.util.Set;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsPathway;
+import org.mobilitydata.gtfsvalidator.table.GtfsPathwayMode;
+import org.mobilitydata.gtfsvalidator.table.GtfsPathwayTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+
+/**
+ * Checks that {@code stops.level_id} is provided when pathways.txt uses {@code
+ * GtfsPathwayMode.ELEVATOR}. This is an implicit check that levels.txt is provided when pathways
+ * uses {@code GtfsPathwayMode.ELEVATOR}. Foreign key validation of {@code pathways.level_id} is
+ * performed in {@code GtfsStopLevelIdForeignKeyValidator}
+ *
+ * <p>Generated notice: {@link MissingLevelIdNotice}.
+ */
+@GtfsValidator
+public class MissingLevelIdValidator extends FileValidator {
+
+  private final GtfsPathwayTableContainer pathways;
+  private final GtfsStopTableContainer stops;
+
+  @Inject
+  MissingLevelIdValidator(GtfsPathwayTableContainer pathways, GtfsStopTableContainer stops) {
+    this.stops = stops;
+    this.pathways = pathways;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    Set<String> reported = new HashSet<>();
+    for (GtfsPathway pathway : pathways.getEntities()) {
+      if (!pathway.pathwayMode().equals(GtfsPathwayMode.ELEVATOR)) {
+        continue;
+      }
+      GtfsStop origin = stops.byStopId(pathway.fromStopId());
+      GtfsStop destination = stops.byStopId(pathway.toStopId());
+      reportStop(noticeContainer, origin, reported);
+      reportStop(noticeContainer, destination, reported);
+    }
+  }
+
+  private static void reportStop(
+      NoticeContainer noticeContainer, GtfsStop stop, Set<String> reported) {
+    if (stop == null) {
+      return;
+    }
```

Comment:
This is a very good check! I am actually thinking of returning Optional from byXyz() functions to avoid NullPointerException. And we can also consider to rewrite the Validator into Kotlin (we can translate Java to Kotlin automatically) to avoid NullPointerException completely.

Reply:
> I am actually thinking of returning Optional from byXyz() functions to avoid NullPointerException. 

I agree that this would be a good improvement!

> And we can also consider to rewrite the Validator into Kotlin (we can translate Java to Kotlin automatically) to avoid NullPointerException completely.

@isabelle-dr @barbeau for future reference.

Reply:
👍 for transition to Kotlin. I've been doing this with most of my Java projects. Automatic translation isn't perfect and requires some manual changes but it's a good start. You can also transition class-by-class, so you don't need to transition the entire project at the same time. IMHO model classes are a good place to start, as they are relatively straightforward and it reduces a lot of boilerplate code.